mantidproject/mantid
# Add summation in Q to ILL's reflectometry reduction

This PR adds a previously missing 'SumInQ' option to `ReflectometryILLSumForeground`. The option enables summation in Q, also know as coherent summation.

There were a few modifications to the overal reduction workflow. Most importantly cropping in wavelength was moved to the end of `ReflectometryILLSumForeground` from `ReflectometryILLPreprocess`.

The wavelength ranges were removed from `ReflectometrySumInQ` and an option to exclude partial bins following #22541 was added.

The changes to algorithm properties and thus the API should not affect any users since none of the algorithms touched here have seen any use outside the development team and test user at ILL.

**Report to:** nobody.

**To test:**

The following script could be used as a starting point for testing. Note that the default facility should be set to ILL and the unit test data directories should be included in Mantid's data search directories before running the script.

```python
ReflectometryILLPreprocess(
    Run='ILL/D17/317369.nxs',
    OutputWorkspace='direct-raw',
    ForegroundHalfWidth=[3],
    OutputBeamPositionWorkspace='direct-beamPos')
ReflectometryILLPreprocess(
    Run='ILL/D17/317370.nxs',
    OutputWorkspace='reflected-raw',
    ForegroundHalfWidth=[3, 4],
    BeamPositionWorkspace='direct-beamPos')
ReflectometryILLSumForeground(
    InputWorkspace='direct-raw',
    OutputWorkspace='direct-fgd')
ReflectometryILLSumForeground(
    InputWorkspace='reflected-raw',
    OutputWorkspace='reflectivity',
    WavelengthRange=[2., 14.],
    SummationType='SumInQ',
    DirectForegroundWorkspace='direct-fgd')
```

Fixes #22315.

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
